src/audio/tdfb/tdfb_ipc3.c

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2023 Intel Corporation. All rights reserved.
+//
+// Author: Baofeng Tian <baofeng.tian@intel.com>
+
+/* this file contains ipc3 specific functions for tdfb.
+ */
+
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/component.h>
+#include <rtos/alloc.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <ipc/header.h>
+#include <ipc/control.h>
+#include <sof/ipc/msg.h>
+#include <errno.h>
+#include <sof/audio/data_blob.h>
+#include <sof/trace/trace.h>
+#include <stdint.h>
+
+#include "tdfb.h"
+#include "tdfb_comp.h"
+
+LOG_MODULE_DECLARE(tdfb, CONFIG_SOF_LOG_LEVEL);
+
+static int init_get_ctl_ipc(struct processing_module *mod)
+{
+	struct tdfb_comp_data *cd = module_get_private_data(mod);
+	int comp_id = dev_comp_id(mod->dev);
+
+	cd->ctrl_data = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, TDFB_GET_CTRL_DATA_SIZE);
+	if (!cd->ctrl_data)
+		return -ENOMEM;
+
+	cd->ctrl_data->rhdr.hdr.cmd = SOF_IPC_GLB_COMP_MSG | SOF_IPC_COMP_GET_VALUE | comp_id;
+	cd->ctrl_data->rhdr.hdr.size = TDFB_GET_CTRL_DATA_SIZE;
+	cd->msg = ipc_msg_init(cd->ctrl_data->rhdr.hdr.cmd, cd->ctrl_data->rhdr.hdr.size);
+
+	cd->ctrl_data->comp_id = comp_id;
+	cd->ctrl_data->type = SOF_CTRL_TYPE_VALUE_CHAN_GET;
+	cd->ctrl_data->cmd = SOF_CTRL_CMD_ENUM;
+	cd->ctrl_data->index = CTRL_INDEX_AZIMUTH_ESTIMATE;
+	cd->ctrl_data->num_elems = 0;
+	return 0;
+}
+
+static void send_get_ctl_ipc(struct processing_module *mod)
+{
+	struct tdfb_comp_data *cd = module_get_private_data(mod);
+
+#if TDFB_ADD_DIRECTION_TO_GET_CMD
+	cd->ctrl_data->chanv[0].channel = 0;
+	cd->ctrl_data->chanv[0].value = cd->az_value_estimate;
+	cd->ctrl_data->num_elems = 1;
+#endif
+
+	ipc_msg_send(cd->msg, cd->ctrl_data, false);
+}
+
+int tdfb_sub_init(struct processing_module *mod)
```

Comment:
The new function name is not descriptive, it should describe better what is done here.